gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -31,54 +31,116 @@
 #include "rust-session-manager.h"
 
 namespace Rust {
+
+// very simple bi-directional hashmap
+template <typename K, typename V> class BiMap
```

Comment:
Wouldn't this class fit more into the utils folder ?

Reply:
Yes agreed, but I don't see it being used very often so I've put it here for now. I don't mind moving it if you think it's useful :)

Reply:
Tbh splitting it now would help other people finding it and reusing it. We probably do not want to deter other people to use such a nice tool simply because it requires a quick refactor before using it.